### ISSUE TITLE
Does vmap not work on GroupNorm?

### ISSUE THREAD

**Issue by QiyaoWei** [pytorch/functorch#622]
Say I want to calculate the jacobian for one sample over ten output classification logits like this
```
import torch
import torch.nn as nn
import torch.nn.functional as F
from functorch import vmap

class Layer(nn.Module):
    def __init__(self, n_channels, n_inner_channels, kernel_size=3, num_groups=1):
        super().__init__()
        self.conv1 = nn.Conv2d(n_channels, n_inner_channels, kernel_size, padding=kernel_size // 2, bias=False)
        self.conv2 = nn.Conv2d(n_inner_channels, n_channels, kernel_size, padding=kernel_size // 2, bias=False)
        self.norm1 = nn.BatchNorm2d(n_inner_channels)
        self.norm2 = nn.BatchNorm2d(n_channels)
        self.norm3 = nn.BatchNorm2d(n_channels)
        self.conv1.weight.data.normal_(0, 0.01)
        self.conv2.weight.data.normal_(0, 0.01)
        self.linear = nn.Linear(3*32*32, 10)

    def forward(self, z, x=None):
        if x == None:
            x = torch.zeros_like(z)
        y = self.norm1(F.relu(self.conv1(z)))
        out = self.norm3(F.relu(z + self.norm2(x + self.conv2(y))))
        return self.linear(torch.flatten(out, 1))

x = torch.randn(5, 3, 32, 32)
model = Layer(3,3)

def jacobian(model, x_i, vs):
    output = model(x_i)
    def vjp(v):
        return torch.autograd.grad(output, model.parameters(), v.view_as(output))
    result = vmap(vjp)(vs)
    return torch.cat([g.flatten(1) for g in result], 1).detach()

vs = torch.eye(10)
# This is vmap
model.eval()
temp = jacobian(model, x[:1], vs)

#This is testing vmap output
output = model(x[:1])
jacobian_rows = [torch.autograd.grad(output, model.parameters(), v.view_as(model(x[:1])), retain_graph=True) for v in vs.unbind()]
jacobian = []
for jac in jacobian_rows:
    jacobian.append(torch.cat([g.flatten(0) for g in jac], 0).detach())
jacobian = torch.stack(jacobian)
assert torch.allclose(temp, jacobian)
```

The assertion works as expected. However, I change nothing else other than the norm function
```
import torch
import torch.nn as nn
import torch.nn.functional as F
from functorch import vmap

class Layer(nn.Module):
    def __init__(self, n_channels, n_inner_channels, kernel_size=3, num_groups=1):
        super().__init__()
        self.conv1 = nn.Conv2d(n_channels, n_inner_channels, kernel_size, padding=kernel_size // 2, bias=False)
        self.conv2 = nn.Conv2d(n_inner_channels, n_channels, kernel_size, padding=kernel_size // 2, bias=False)
        self.norm1 = nn.GroupNorm(num_groups, n_inner_channels)
        self.norm2 = nn.GroupNorm(num_groups, n_channels)
        self.norm3 = nn.GroupNorm(num_groups, n_channels)
        self.conv1.weight.data.normal_(0, 0.01)
        self.conv2.weight.data.normal_(0, 0.01)
        self.linear = nn.Linear(3*32*32, 10)

    def forward(self, z, x=None):
        if x == None:
            x = torch.zeros_like(z)
        y = self.norm1(F.relu(self.conv1(z)))
        out = self.norm3(F.relu(z + self.norm2(x + self.conv2(y))))
        return self.linear(torch.flatten(out, 1))

x = torch.randn(5, 3, 32, 32)
model = Layer(3,3)

def jacobian(model, x_i, vs):
    output = model(x_i)
    def vjp(v):
        return torch.autograd.grad(output, model.parameters(), v.view_as(output))
    result = vmap(vjp)(vs)
    return torch.cat([g.flatten(1) for g in result], 1).detach()

vs = torch.eye(10)
# This is vmap
model.eval()
temp = jacobian(model, x[:1], vs)

#This is testing vmap output
output = model(x[:1])
jacobian_rows = [torch.autograd.grad(output, model.parameters(), v.view_as(model(x[:1])), retain_graph=True) for v in vs.unbind()]
jacobian = []
for jac in jacobian_rows:
    jacobian.append(torch.cat([g.flatten(0) for g in jac], 0).detach())
jacobian = torch.stack(jacobian)
assert torch.allclose(temp, jacobian)
```

And the assertion fails. In fact, if we look at the gradients obtained by vmap we find that the first row is correct, and all others are zero. Is this an intended behavior? If not, how would I go about fixing this?

**Comment by Chillee** [comment]
This sounds like it might be a bug in our group norm batching rules. @samdow

**Comment by samdow** [comment]
@QiyaoWei It should be fixed but please do let us know if you're still seeing issues or have other questions. I was able to get your example to now pass `torch.allclose(..., atol=1e-4)`
